crates/nu-protocol/src/value/mod.rs

```diff
@@ -2409,3 +2406,63 @@ fn get_config_filesize_format(config: &Config) -> (ByteUnit, &str) {
 
     filesize_format
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::{Span, Value};
+
+    mod is_empty {
+        use super::*;
+
+        #[test]
+        fn test_string() {
+            let value = Value::string("", Span::unknown());
+            assert!(value.is_empty());
+        }
+
+        #[test]
+        fn test_list() {
+            let list_with_no_values = Value::List {
+                vals: vec![],
+                span: Span::unknown(),
+            };
+            let list_with_one_empty_string = Value::List {
+                vals: vec![Value::string("", Span::unknown())],
+                span: Span::unknown(),
+            };
+
+            assert!(list_with_no_values.is_empty());
+            assert!(!list_with_one_empty_string.is_empty());
+        }
+
+        #[test]
+        fn test_record() {
+            let no_columns_nor_cell_values = Value::Record {
+                cols: vec![],
+                vals: vec![],
+                span: Span::unknown(),
+            };
+            let one_column_and_one_cell_value_with_empty_strings = Value::Record {
```

Comment:
@jntrnr in theory it's impossible to have a record like this, right? Please advise.

Reply:
The one we are assigning to one_column_and_one_cell_value_with_empty_strings

Reply:
Nevermind....

I meant a record with no cols but with vals.